irohad/consensus/yac/impl/yac_crypto_provider_impl.cpp

```diff
@@ -61,19 +63,17 @@ namespace iroha {
         auto signature = shared_model::crypto::CryptoSigner<>::sign(
             blob, shared_model::crypto::Keypair(pubkey, privkey));
 
-        shared_model::builder::DefaultSignatureBuilder()
-            .publicKey(pubkey)
-            .signedData(signature)
-            .build()
-            .match([&vote](iroha::expected::Value<
-                           std::shared_ptr<shared_model::interface::Signature>>
-                               &sig) { vote.signature = sig.value; },
-                   [](iroha::expected::Error<std::shared_ptr<std::string>>
-                          &reason) {
-                     logger::log("YacCryptoProvider::getVote")
-                         ->error("Cannot build vote signature: {}",
-                                 *reason.error);
-                   });
+        factory_->createSignature(pubkey, signature)
+            .match(
+                [&](iroha::expected::Value<
+                    std::unique_ptr<shared_model::interface::Signature>> &sig) {
+                  vote.signature = std::move(sig.value);
+                },
+                [&](iroha::expected::Error<std::string> &reason) {
+                  logger::log("YacCryptoProvider::getVote")
+                      ->error("Cannot build vote signature: {}", reason.error);
```

Comment:
That branch will probably lead to UB, since `vote.signature` is undefined

Reply:
Behavior is defined, it will result in segfault on the attempt to access `vote.signature` in next method, because the default value of shared pointer is nullptr. This pull request is not aimed at fixing the behavior, and since this error is logged, it is possible to fix it in another task.